packages/flutter_tools/test/integration.shard/flutter_build_preview_sdk.dart

```diff
@@ -91,4 +93,40 @@ void main() {
         .childFile('app-debug.apk').existsSync(), true);
     },
   );
+
+  test(
+    'build succeeds when both example app and plugin target compileSdkPreview',
+        () async {
+      final File appBuildGradleFile = exampleAppDir.childDirectory('android').childDirectory('app').childFile('build.gradle');
+      // write a build.gradle with compileSdkPreview as `Tiramisu` which is a string preview version
+      appBuildGradleFile.writeAsStringSync(
+          appBuildGradleFile.readAsStringSync().replaceFirst('compileSdkVersion flutter.compileSdkVersion', 'compileSdkPreview "UpsideDownCake"'),
+          flush: true
+      );
+      expect(appBuildGradleFile.readAsStringSync(), contains('compileSdkPreview "Tiramisu"'));
```

Comment:
From what I can tell this is because the tools tests filter to [only files that end in `_test.dart`](https://github.com/flutter/flutter/blob/1cf3907407cbc91be7cec2c38b348a2d66041dd5/dev/bots/test.dart#L461), presumably to avoid including things [like `test_utils.dart`](https://github.com/flutter/flutter/blob/1cf3907407cbc91be7cec2c38b348a2d66041dd5/packages/flutter_tools/test/integration.shard/test_utils.dart), so it was never actually running. I've renamed to allow it to run.

Reply:
and it looks like they all fail 🤦 . I'll convert to draft until they're fixed